packages/stryker-webpack-transpiler/src/compiler/WebpackCompiler.ts

```diff
@@ -0,0 +1,66 @@
+import { TextFile, File, BinaryFile, FileKind } from 'stryker-api/core';
+import { Compiler, Configuration } from 'webpack';
+import webpack from './Webpack';
+import * as path from 'path';
+import InputFileSystem from '../fs/InputFileSystem';
+import OutputFileSystem from '../fs/OutputFileSystem';
+
+export default class WebpackCompiler {
+  private _compiler: Compiler;
+  private _inputFS: InputFileSystem;
+  private _outputFS = new OutputFileSystem();
```

Comment:
Can't you pass an initialized InputFileSystem in the constructor, thereby using dependency injection? That would also make it easier to mock away the filesystem in the unit tests.

Reply:
I know this is possible using sandboxes but the cleaner way would be injecting them in the constructor.

Reply:
I agree that constructor params are more explicit. I'll change it to:

```ts
  public constructor(webpackConfig: Configuration,
    private _inputFS = new InputFileSystem(),
    private _outputFS = new OutputFileSystem()) {
    this._compiler = this.createCompiler(webpackConfig);
  }
```